includes/Core/Storage/Setting_With_Legacy_Keys_Trait.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Trait Google\Site_Kit\Core\Storage\Setting_With_Legacy_Keys_Trait
+ *
+ * @package   Google\Site_Kit\Core\Storage
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Storage;
+
+use \Google\Site_Kit\Core\Util\Migrate_Legacy_Keys;
+
+/**
+ * Trait for a Setting that has legacy option keys to migrate.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+trait Setting_With_Legacy_Keys_Trait {
+	use Migrate_Legacy_Keys;
+
+	/**
+	 * Mapping of legacy keys to current key.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array
+	 */
+	abstract protected function get_legacy_key_map();
+
+	/**
+	 * Registers an option filter for the setting to migrate legacy keys.
+	 *
+	 * @since n.e.x.t
+	 */
+	protected function add_legacy_key_migration_filters() {
```

Comment:
Maybe we could have this accept the `$legacy_key_map` as parameter instead, it would make this method more declarative, and we wouldn't need to implement another abstract method in each class using this.

Reply:
Sure, I suppose we could do that. I'm not sure it makes it much more declarative using a parameter, but it would save the need for an added method.

Reply:
Given the abstract trait method conversation, I think we can keep this. It's not strictly necessary to have a property since this is a one-off value, hence just passing a parameter might still make sense. But I'll leave that to you.

Reply:
I've reworked this to use a parameter instead which is less code/boilerplate than a property or a method so I think this is a win. I also renamed the method to `register_legacy_keys_migration( array $legacy_key_map )` which reads much more naturally I think with the parameter than before.